### DISCUSSION TITLE
Authentication Failed

### DISCUSSION THREAD

**Discussion by ss0703** [fredemmott/StreamDeck-Discord#67] [Q&A]
I've added the application id and public key several times. I've even deleted the app and created new ones and it still shows AUTHENTICATION_FAILED on the debug information. 

Discord is current version and NOT running in administrator mode.

https://localhost/ is set as redirect.

Anyone else having issues?

**Comment by fredemmott** [comment]
The plugin requires the secret/private key; have you tried this?

**Reply by ss0703** [reply]
Weird... for my Discord Developer Portal I have Application ID and Public Key (which I'm guessing they've changed?). 

But either way, those are the things I copy paste into it and still get AUTHENTICATION_FAILED

**Reply by High-C3PO** [reply]
Thanks @fredemmott I was copy/pasting the wrong credentials into the plugin. Not OP but this saved me. @ss0703 you need to go into the /OAuth2 tab of your Discord app and copy/paste the client ID and Secret. It sounds like you're having the same issue as myself caused by using the Application ID and Public Key from the /General Information tab.

**Reply by fredemmott** [reply]
Thanks, I'll add that into the TROUBLESHOOTING docs :)

**Reply by alecbramham** [reply]
I'm also having this problem. In the Developer Portal app I have the redirect set.
When I copy the client ID & Secret into the Stream Deck app, the debug menu will say "Requesting Authentication". After accepting on Discord once, "Authentication Failed" shows up in debug. I don't get asked a second time. Pressing the buttons on the streamdeck gives me a yellow exclamation mark.

The authentication window looks different, but I think that is just a side effect of the video being one year old.
